### PR TITLE
For #24618 - Support clicking on web notifications

### PR DESCRIPTION
**Pull request by Mugurell**
Needs https://github.com/mozilla-mobile/android-components/pull/11957

Tested on https://cleverpush.com/en/test-notifications/

https://user-images.githubusercontent.com/11428869/161807042-00c98bc4-30ca-4414-8438-58ccb76a5890.mp4

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/IntentProcessorType.kt [hunk 1]
```diff
@@ -40,7 +40,8 @@ fun IntentProcessors.getType(processor: IntentProcessor?) = when {
         privateCustomTabIntentProcessor == processor -> IntentProcessorType.EXTERNAL_APP
     intentProcessor == processor ||
         privateIntentProcessor == processor ||
-        fennecPageShortcutIntentProcessor == processor -> IntentProcessorType.NEW_TAB
+        fennecPageShortcutIntentProcessor == processor ||
+        webNotificationsIntentProcessor == processor -> IntentProcessorType.NEW_TAB
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/components/IntentProcessorType.kt:44]: this seems like a somewhat complex `when`. should we get a test for it?

File: app/src/main/java/org/mozilla/fenix/ServiceWorkerSupportFeature.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * which adds the ability to navigate to the browser before opening a new tab.
+ *
+ * Will automatically register callbacks for service workers requests and cleanup when [homeActivity] is destroyed.
+ *
+ * @param homeActivity [HomeActivity] used for navigating to browser or accessing various app components.
+ */
+class ServiceWorkerSupportFeature(
+    private val homeActivity: HomeActivity
+) : ServiceWorkerDelegate, DefaultLifecycleObserver {
+    override fun onDestroy(owner: LifecycleOwner) {
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/ServiceWorkerSupportFeature.kt:26]: I left comments on the A-C side. This is a bit of a new pattern where we have a global delegate tied to the application lifecycle but also need a reference to our activity. I think this solution is fine, but some changes over at A-C needed.

Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/ServiceWorkerSupportFeature.kt:26]: Updated the AC patch. Thank you!

File: app/src/main/java/org/mozilla/fenix/components/IntentProcessorType.kt (hunk 1 above)
Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/components/IntentProcessorType.kt:44]: Tests for checking the returned `IntentProcessorType` existed already. Added a new one for the new processor in `IntentProcessorTypeTest`. Thank you!

**Review by csadilek — APPROVED** [review]

**Comment by Mugurell** [comment]
Unrelated test failing.
Restarting CI.

```
TEST: GIVEN tab with search terms WHEN subsequent direct load occurs THEN search terms are not retained
...
FAILURE

org.mockito.exceptions.misusing.InvalidUseOfMatchersException:
Misplaced or misused argument matcher detected here:
...
Mocking methods declared on non-public parent classes is not supported.
at org.mozilla.fenix.home.recentvisits.HistoryMetadataMiddlewareTest.setupGoogleSearchEngine(HistoryMetadataMiddlewareTest.kt:1101)
at org.mozilla.fenix.home.recentvisits.HistoryMetadataMiddlewareTest.GIVEN tab with search terms WHEN subsequent direct load occurs THEN search terms are not retained(HistoryMetadataMiddlewareTest.kt:309)
```

**Comment by Mugurell** [comment]
Saw the test failing again, could be because `HistoryMetadataMiddlewareTest.setupGoogleSearchEngine` was using a `mozilla.components.support.test.mock`.
Replaced that with a `io.mockk.mockk`. 🤞

**Comment by Mugurell** [comment]
Same issue now for `org.mozilla.fenix.telemetry.TelemetryMiddlewareTest.WHEN background tab gets killed THEN middleware records background age`

**Pull request merged**